lib/search_terms/keyword.rb

```diff
@@ -1,8 +1,56 @@
 class Keyword
-  def initialize(string)
+
+  module SensitiveSearchStrategy
+    def matches?(line)
+      line.include? @string
+    end
+  end
+
+  module InsensitiveSearchStrategy
+    def matches?(line)
+      line.downcase.include? @string.downcase
+    end
+  end
+
+  def initialize(string,mode = :sensitive)
+    strategize(mode)
     @string = string
   end
-  def matches?(line)
-    line.include?(@string)
+
+  private
+
+  def strategize(mode)
+    # capitalize the first letter of the mode
+    strategy = mode.slice(0,1).capitalize + mode.slice(1..-1)
```

Comment:
In Rails-land we can use the [constantize](http://api.rubyonrails.org/classes/ActiveSupport/Inflector.html#method-i-constantize) method to reduce this to a one-liner. We would need to include the ActiveSupport gem for that, which I generally avoid doing in non-Rails projects.

```ruby
  def strategize(mode)
    extend "Keyword::#{mode}SearchStrategy".constantize
  end
```